### PR TITLE
[DOCS] Adds classification type DFA API docs and ml-shared.asciidoc

### PR DESCRIPTION
**Pull request by szabosteve**
This PR adds the API documentation for the classification type data frame analytics and an ml-shared.asciidoc file that contains common DFA variables.

Related issue: https://github.com/elastic/stack-docs/issues/629

### CODE REVIEW

File: docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc [hunk 1]
```diff
@@ -180,22 +181,17 @@ hyperparameter optimization to give minimum validation errors.
 ===== Standard parameters
 
 `dependent_variable`::
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc]: Don't we want to reuse the definition from ml-shared?

File: docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc [hunk 2]
```diff
@@ -209,46 +205,75 @@ values unless you fully understand the function of these parameters. If these
...
- (Optional, double) Regularization parameter to prevent overfitting on the 
- training dataset. Multiplies a linear penalty associated with the size of 
- individual trees in the forest. The higher the value the more training will 
- prefer smaller trees. The smaller this parameter the larger individual trees 
- will be and the longer train will take.
+[float]
+[[classification-resources-standard]]
+===== Standard parameters
+ 
+`dependent_variable`::
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc]: Don't we want to reuse the definition from ml-shared?

File: docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc]: @dimitris-athanasiou The definition of the `dependent_variable` is not identical for regression (only numeric field) and for classification (numeric or boolean field), that's why I eventually didn't use the definition from ml-shared. A possible workaround came to my mind, my last commit added it.

File: docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc (hunk 2 above)
Comment (szabosteve) [reply docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc]: See above.

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Comment by szabosteve** [comment]
@elasticmachine teest this pls.

**Comment by szabosteve** [comment]
@elasticmachine test this pls

**Comment by szabosteve** [comment]
@elasticmachine test this please

**Pull request merged**